commons/commons-util/src/main/kotlin/ebi/ac/uk/util/file/PoiExt.kt

```diff
@@ -27,7 +26,21 @@ private fun Row.asString(): String {
         cells.add(getCell(idx)?.valueAsString ?: EMPTY)
     }
 
-    return cells.joinToString("\t")
+    return when {
+        cells.all { it == EMPTY } -> EMPTY
+        else -> cells.trim().joinToString("\t")
+    }
+}
+
+private fun MutableList<String>.trim(): MutableList<String> {
+    if (last().isBlank()) {
+        for (idx in indices.reversed()) {
+            if (elementAt(idx).isNotBlank()) break
+            removeAt(idx)
+        }
+    }
```

Comment:
this for looks overcomplex, definitely kotlin have a cleaner declarative approach. What about

```
this.asReversed().dropWhile { it.isBlank() }
```

Reply:
Thanks! I knew there was a better way. You're like my Kotlin linter :laughing: 

Only one detail, I ended up using: `dropLastWhile { it.isBlank() }`